command/command.ts

```diff
@@ -445,6 +450,7 @@ export class Command<
     } else if (typeof help === "function") {
       this.cmd._help = help;
     } else {
+      this.exitOnHelp = help.exit ?? true;
```

Comment:
i think this wan't work with sub-commands. the `exitOnHelp` must be set on the current selected command and we need a function that checks if the current or any parent command has this option enabled or disable.
```suggestion
      this.cmd.exitOnHelp = help.exit ?? true;
```